### DISCUSSION TITLE
Query Foreign Table in Stream

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#5810] [Questions] [answered]
I'm looking for a way to query foreign tables in the stream, with some foreign key data
like for the case:

![image](https://user-images.githubusercontent.com/44422128/156879721-ed0a39c6-fa37-4675-8345-0a7a11969881.png)

I have a 'Rooms' table with the following fields, and the sender field is a foreign key here, but there's no documentation for querying a foreign table in a stream. and following code returns the response as shown in the image below the code.

    Stream<List<Map<String, dynamic>>> get_chat_rooms() {
    final res = supabaseClient
        .from(
            'Rooms:location_id=eq.${appLocationController.currentLocation.value.id}')
        .stream(['Users:sender(*)']).execute();
    return res;  }
![image](https://user-images.githubusercontent.com/44422128/156879771-96201f12-f4e1-4331-b37e-b1a3e9780aaf.png)

the 'sender' should be a user object like normal SQL queries with joins. 
following shows what currently object is like and how it should be:

![image](https://user-images.githubusercontent.com/44422128/156880361-9f2924bb-21ca-41d9-8e80-02a1e6d1d24c.png)

**Comment by GaryAustin1** [comment, marked as answer]
Realtime, and Stream which uses it, only return row data for the monitored table/row.  The array of ids in .stream is used to put updates and inserts into the stream(list) in the correct spot. .  Because the stream handler just uses the payload from realtime (and not a separate select like you would do for a foreign key operation) it can’t get data beyond the monitored table row.
